### PR TITLE
Code consolidation

### PR DESCRIPTION
**Pull request by JoseCastilloExperian**
Refer to task #147271

### CODE REVIEW

**Comment by JoseCastilloExperian** [comment]
Some of the tests weren't running properly, already fix them and all of them pass correctly.

File: tests/tests/functional SFRA/Checkout/integration.js [hunk 1]
```diff
@@ -220,12 +230,14 @@ registerSuite('Edq Cartridge Functional Test 2', {
...
+					.end()
+				.sleep(5000)
 				.findByName('dwfrm_billing_contactInfoFields_email')
 					.clearValue()
 					.type("noreply@gmail.com")
-					.sleep(500)
-				.findByCssSelector("#form-submit")
+					.end()
+				.sleep(500)
+				.findByXpath('//button[@name="submit" and @value="submit-payment"]')
```
[the remark this answers is not in the document]
Comment (JoseCastilloExperian) [reply tests/tests/functional SFRA/Checkout/integration.js:225]: When I start building the tests; I was having an error that says that the element was unreachable (can't remember the exact error), when I started to use XPath.

File: tsSource/src/utils/edq-utils.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	fn: any;
+}
+/**
+ * Add event listeners to the elements.
+ *
+ * @param {element} selector
+ * @param {event} event
+ * @param {function} fn
+ */
+export function addEventOnElement({selector, event, fn}: UtilsAddEventsArgs) {
```
[the remark this answers is not in the document]
Comment (JoseCastilloExperian) [reply tsSource/src/utils/edq-utils.ts:35]: Yes, noticed that some elements were using manual attachments; already changed to use the function.

File: tsSource/src/sfra.ts [hunk 3]
```diff
@@ -0,0 +1,266 @@
+/*global EDQ*/
+/*eslint no-undef: "error"*/
+/*eslint no-unused-vars: ["error", { "vars": "local" }]*/
+/* exported edqSetEmailValidationConfiguration edqSetPhoneValidationConfiguration setCheckoutFormEvents edqValidateAddressCallBack edqCheckoutPageWorkflows */
+export {};
```
[the remark this answers is not in the document]
Comment (JoseCastilloExperian) [reply tsSource/src/sfra.ts]: Forgot to remove it; already remove it.

File: int_experian/int_experian_sfra/cartridge/templates/default/edq/globalIntuitive.isml [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
-    edqSetGlobalIntuitiveConfiguration();
-    setCheckoutFormEvents();
+	window.EdqDemandware.sfra.useGlobalIntuitive(vDefaultCountry, edqAuthorizationToken, edqDataSetUsage, edqDataSetCode);
 </script>
 <isif condition="${dw.system.Site.current.preferences.custom.EDQStaging == 'Development'}">
     <isloop items="${dw.object.SystemObjectMgr.describe('SitePreferences').getCustomAttributeDefinition('EDQDevUrls').values}" var="dev">
         <isif condition="${dev.value == 'GlobalIntuitiveJS'}">
         	<script src="${dev.displayValue}" integrity="sha512-ooVQYWcrVoGAZXC+qPMJaFsEkLB82EsT42J+p0U5INWm+NHrN+XvnWsALGv440Zg3a/QsDJ2L9XR0592fkSBmA==" crossorigin="anonymous"></script>
         	<script type="text/javascript">
+				edqGlobalIntuitiveStagging = false;
```
[the remark this answers is not in the document]
Comment (JoseCastilloExperian) [reply int_experian/int_experian_sfra/cartridge/templates/default/edq/globalIntuitive.isml]: Ohhh, I made the correction!!😄

File: tests/tests/functional SGJC/Registration/integration.js [hunk 5]
```diff
@@ -76,22 +76,19 @@ function fillInPartialAddress() {
...
-				.end()
-		},	
+				.sleep(2000)
+				.find("css selector", ".ui-button-text-only")
+					.click()
+					.end()
+				.catch(function(errorLog) { })
+		},
 		tests: {
 		"SGJC Create Account - Email Validation Succed": function() {
```
[the remark this answers is not in the document]
Comment (JoseCastilloExperian) [reply tests/tests/functional SGJC/Registration/integration.js]: Yes!!

File: tsSource/src/utils/edq-utils.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
+import {edqEmailValidationCallback} from './edqEmailValidation';
+import {edqPhoneValidationCallback} from './edqPhoneValidation';
+import {countryAlpha3} from './edqCountry';
+/**
+ * In the Business Manager there's an option that sets if the email and/or email will allow the user to prevent the
```
[the remark this answers is not in the document]
Comment (JoseCastilloExperian) [reply tsSource/src/utils/edq-utils.ts]: Meant to say "phone and/or email".

File: int_experian/int_experian_sgjc/cartridge/templates/default/edq/edqHeaders.isml [hunk 7]
```diff
@@ -8,7 +8,7 @@
             <link rel="stylesheet" href="${dev.displayValue}" type="text/css" />
         </isif>
         <isif condition="${dev.value == 'EDQJS'}">
-			<script src="${dev.displayValue}" integrity="sha512-L23hwvucqXSQX9r/bCvFq92VmCXUJqPfpvjwlP0AxFUtLR304o1U1RcGrgfiv0S83bt7s+T+0m82JFEIyPAH3g==" crossorigin="anonymous"></script>
+            <script src="${dev.displayValue}" integrity="sha512-L23hwvucqXSQX9r/bCvFq92VmCXUJqPfpvjwlP0AxFUtLR304o1U1RcGrgfiv0S83bt7s+T+0m82JFEIyPAH3g==" crossorigin="anonymous"></script>
```
Comment (tabeth) [inline int_experian/int_experian_sgjc/cartridge/templates/default/edq/edqHeaders.isml:11]: What we'll do is setup the build so these integrity checks are dynamically generated. For now this is fine, though.

File: tests/tests/functional SFRA/Checkout/integration.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tabeth) [reply tests/tests/functional SFRA/Checkout/integration.js:225]: I don't understand - are you saying the element is unreachable if you're using XPaths or when you're using selectors?

File: tests/tests/functional SFRA/Registration/integration.js [hunk 8]
```diff
@@ -2,99 +2,26 @@ const SfraLoginUrl = "https://qas01.tech-prtnr-na01.dw.demandware.net/on/demandw
...
-				.end()
-	}
-}
-*/
-registerSuite('Edq Cartridge Functional Test 3', {
+registerSuite('Experian SFRA Registration Touchpoint', {
 	before: function() {
 		return this.remote
-			.setFindTimeout(10000)
+			//.setFindTimeout(500)
```
Comment (tabeth) [inline tests/tests/functional SFRA/Registration/integration.js:9]: Remove commented out code.

**Pull request merged**